### PR TITLE
Added license info to rest.md

### PR DESCRIPTION
**Pull request by chipaca**
also: bumped api_compat; added json tags to licenseData, and added an integration-ish test for it all

### CODE REVIEW

**Comment by mvo5** [comment]
Looks good, +1

File: daemon/api_test.go [hunk 1]
```diff
@@ -1151,3 +1152,45 @@ func (s *apiSuite) TestInstallLicensed(c *check.C) {
...
+
+	req, err := http.NewRequest("POST", "/1.0/packages/foo.bar", strings.NewReader(`{"action": "install"}`))
+	c.Assert(err, check.IsNil)
+	s.vars = map[string]string{"name": "foo", "origin": "bar"}
+
+	res := postPackage(packageCmd, req).(*resp).Result.(map[string]interface{})
+	task := d.tasks[res["resource"].(string)[16:]]
+	c.Check(task, check.NotNil)
+
+	task.tomb.Wait()
```
Comment (niemeyer) [inline daemon/api_test.go]: Is the error worth checking here?

File: daemon/api_test.go [hunk 2]
```diff
@@ -1151,3 +1152,45 @@ func (s *apiSuite) TestInstallLicensed(c *check.C) {
...
+	})
+
+	req, err = http.NewRequest("POST", "/1.0/packages/foo.bar", strings.NewReader(`{"action": "install", "license": {"intro": "hi", "license": "yak yak", "agreed": true}}`))
+	c.Assert(err, check.IsNil)
+
+	res = postPackage(packageCmd, req).(*resp).Result.(map[string]interface{})
+	task = d.tasks[res["resource"].(string)[16:]]
+	c.Check(task, check.NotNil)
+
+	task.tomb.Wait()
```
Comment (niemeyer) [inline daemon/api_test.go]: Same.. if it fails, would this give insight?

**Comment by niemeyer** [comment]
LGTM. A couple of trivial ponderings which may be done in a follow up, if relevant.

**Pull request merged**